scanpy/plotting/_anndata.py

```diff
@@ -1790,64 +1792,39 @@ def _prepare_dataframe(
     -------
     Tuple of `pandas.DataFrame` and list of categories.
     """
-    from scipy.sparse import issparse
 
     sanitize_anndata(adata)
     use_raw = _check_use_raw(adata, use_raw)
+    if layer is not None:
+        use_raw = False
     if isinstance(var_names, str):
         var_names = [var_names]
 
+    orig_index_col = None
     if groupby is not None:
         if isinstance(groupby, str):
             # if not a list, turn into a list
             groupby = [groupby]
         for group in groupby:
+            if group == adata.obs.index.name:
```

Comment:
This should error if `group` is both the index name and a column name, like pandas does:

```python
pd.DataFrame({"a": [1,2,3]}, index=pd.Index(list("abc"), name="a")).groupby("a")
```

```pytb
ValueError: 'a' is both an index level and a column label, which is ambiguous.
```

Reply:
done